### PR TITLE
make new nodegroups inherit control plane version

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

- this allow two new modes:
  - `--version=auto` (new default) will inherit version of CP
  - `--version=latest` forces use of latest known version

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [ ] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] All integration tests passing (i.e. `make integration-test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)

### CODE REVIEW

**Comment by errordeveloper** [comment]
cc @whereisaaron @christopherhein

**Comment by errordeveloper** [comment]
I was wondering if we should add a check that go over existing nodegroups
and warns user if version drift is detected. Maybe another PR would be
best, but this looks like a good place to discuss it!

**Comment by christopherhein** [comment]
> I was wondering if we should add a check that go over existing nodegroups and warns user if version drift is detected. Maybe another PR would be best, but this looks like a good place to discuss it!

I would say yes, didn't we have another issue that was about this type of check system? Maybe I'm confusing the one with checking if upgrades would be do-able given the state of CFN.

**Comment by whereisaaron** [comment]
A drift warning seems an out of scope bell/whistle for this PR. But a PR for a `eksctl utils cluster-health` that checked versions and maybe things like ASG health and node status would be great.

Would it be possible to display the k8s version as a column in the `eksctl get nodegroup` output? Then you can see at a glance if there are differences across the node groups. You can kinda do that now with the AMI. 

```
$ if [[ $(eksctl get nodegroup --cluster=cave -o json | jq ".[].ImageID" | sort | uniq -u | wc -l) -gt 1 ]]; then echo 'DRIFT!'; else echo 'ALL GOOD!'; fi
ALL GOOD!
```

**Comment by errordeveloper** [comment]
@christopherhein yes, you are thinking of #539.

@whereisaaron you are right, it's probably a good idea to add a separate command that checks state of cluster configuration overall. 

> Would it be possible to display the k8s version as a column in the `eksctl get nodegroup` output?

No, at the moment we don't store version information directly, we'd have ask Kubernetes for it. You can actually already do this yourself using `kubectl`. If it's the AMI that you want to know, I don't recall if that gets tracked by Kubernetes at all, but version of kubelet seems most relevant really.

File: pkg/ctl/create/nodegroup.go [hunk 1]
```diff
@@ -96,6 +98,41 @@ func filterNodeGroups(cfg *api.ClusterConfig) error {
 	return nil
 }
 
+func checkVersion(ctl *eks.ClusterProvider, meta *api.ClusterMeta) error {
```
Comment (errordeveloper) [inline pkg/ctl/create/nodegroup.go]: Ended up with a simple function here to begin with, we might want to find a better place for it later. I think we might actually promote nodegroup-level version field to reflect the real world, so I'm not too worried about generalising this.

**Pull request merged**